src/platforms/hosted/jlink.c

```diff
@@ -303,61 +567,77 @@ bool jlink_init(void)
 		libusb_close(info.usb_link->device_handle);
 		return false;
 	}
-	jlink_info();
+	if (!jlink_get_capabilities() || !jlink_get_version() || !jlink_get_interfaces()) {
+		DEBUG_ERROR("Failed to read J-Link information\n");
+		libusb_release_interface(info.usb_link->device_handle, info.usb_link->interface);
+		libusb_close(info.usb_link->device_handle);
+		return false;
+	}
+	strncpy(info.version, jlink.fw_version, sizeof(info.version));
```

Comment:
Rather than using strncpy() which will write past the end of the used part of the version string with a filler character, please use memcpy() here as you already guarantee NUL character termination when reading the firmware version string.

```suggestion
	memcpy(info.version, jlink.fw_version, strlen(jlink.fw_version) + 1U);
```

Reply:
good catch